java/main/jglome/Glome.java

```diff
@@ -0,0 +1,357 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jglome;
+
+import static java.lang.System.arraycopy;
+import static java.security.spec.NamedParameterSpec.X25519;
+
+import java.nio.ByteBuffer;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.interfaces.XECPrivateKey;
+import java.security.interfaces.XECPublicKey;
+import java.security.spec.NamedParameterSpec;
+import java.util.Arrays;
+import java.util.Optional;
+import javax.crypto.KeyAgreement;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Class, which encapsulates the logic of GLOME protocol.
+ */
+public class Glome {
+
+  private XECPublicKey peerKey;
+  private Mac userMacKey;
+  private Mac peerMacKey;
+  private int minPeerTagLength;
+
+  private final KeyPair userKeys;
+
+  private final static int MIN_CNT_VALUE = 0;
+  private final static int MAX_CNT_VALUE = 255;
+  private final static int MAX_TAG_LENGTH = 32;
+  private final static int MIN_TAG_LENGTH = 1;
+
+  /**
+   * First Glome constructor. Generates a pair of private/public keys for the user, calculates
+   * user's and peer's MAC keys.
+   *
+   * @param peerKey peer's public key.
+   * @param minPeerTagLength minimum allowed peer's tag length.
+   * @throws MinPeerTagLengthOutOfBoundsException if {@code minPeerTagLength} is out of [1..32]
+   * range.
+   */
+  public Glome(XECPublicKey peerKey, int minPeerTagLength)
+      throws MinPeerTagLengthOutOfBoundsException {
+    this.userKeys = generateKeys();
+
```

Comment:
I'd remove the newline between this and initMacKeys (also below). "A single blank line may also appear anywhere it improves readability, for example between statements to organize the code into logical subsections." I'd say that these constructors are sufficiently small that they don't need the additional vertical whitespace.